File: api/editor/api-editor/src/main/java/ca/bc/gov/tno/areas/editor/models/TagModel.java
package ca.bc.gov.tno.areas.editor.models;

import ca.bc.gov.tno.dal.db.entities.Tag;
import ca.bc.gov.tno.models.AuditColumnModel;

public class TagModel extends AuditColumnModel {
  /**
   * Primary key to identify the tag.
   */
  private String id;

  /**
   * A unique name to identify the tag.
   */
  private String name;

  /**
   * A description of the tag.
   */
  private String description = "";

  /**
   * The order to display.
   */
  private int sortOrder;

  /**
   * Whether this record is enabled or disabled.
   */
  private boolean enabled = true;

  /**
   * A collection of role categories that belong to this tag.
   */
  // private List<ContentTag> contentCategories = new ArrayList<>();

  public TagModel() {
  }

  public TagModel(Tag entity) {
    super(entity);

    if (entity != null) {
      this.id = entity.getId();
      this.name = entity.getName();
      this.description = entity.getDescription();
      this.enabled = entity.isEnabled();
      this.sortOrder = entity.getSortOrder();
    }
  }

  /**
   * @return String the id
   */
  public String getId() {
    return id;
  }

  /**
   * @param id the id to set
   */
  public void setId(String id) {
    this.id = id;
  }

  /**
   * @return String the name
   */
  public String getName() {
    return name;
  }

  /**
   * @param name the name to set
   */
  public void setName(String name) {
    this.name = name;
  }

  /**
   * @return String the description
   */
  public String getDescription() {
    return description;
  }

  /**
   * @param description the description to set
   */
  public void setDescription(String description) {
    this.description = description;
  }

  /**
   * @return boolean the enabled
   */
  public boolean isEnabled() {
    return enabled;
  }

  /**
   * @param enabled the enabled to set
   */
  public void setEnabled(boolean enabled) {
    this.enabled = enabled;
  }

  /**
   * @return int the sortOrder
   */
  public int getSortOrder() {
    return sortOrder;
  }

  /**
   * @param sortOrder the sortOrder to set
   */
  public void setSortOrder(int sortOrder) {
    this.sortOrder = sortOrder;
  }

}

File: libs/net/models/Areas/Admin/Tags/TagModel.cs
using TNO.API.Models;

namespace TNO.API.Areas.Admin.Models.Tag;

/// <summary>
/// TagModel class, provides a model that represents an media type.
/// </summary>
public class TagModel : AuditColumnsModel
{
    #region Properties
    /// <summary>
    /// get/set - The primary key of the type model.
    /// </summary>
    public int Id { get; set; }
    
    /// <summary>
    /// get/set - Unique name to identify the entity.
    /// </summary>
    public string Code { get; set; } = "";

    /// <summary>
    /// get/set - The unique name of the model.
    /// </summary>
    public string Name { get; set; } = "";

    /// <summary>
    /// get/set - A description of the type model.
    /// </summary>
    public string Description { get; set; } = "";

    /// <summary>
    /// get/set - Whether this model is enabled.
    /// </summary>
    public bool IsEnabled { get; set; }

    /// <summary>
    /// get/set - The sort order of the models.
    /// </summary>
    public int SortOrder { get; set; }
    #endregion

    #region Constructors
    /// <summary>
    /// Creates a new instance of an TagModel.
    /// </summary>
    public TagModel() {}

    /// <summary>
    /// Creates a new instance of an TagModel, initializes with specified parameter.
    /// </summary>
    /// <param name="entity"></param>
    public TagModel(Entities.Tag entity) : base(entity)
    {
        this.Id = entity.Id;
        this.Name = entity.Name;
        this.Code = entity.Code;
        this.Description = entity.Description;
        this.SortOrder = entity.SortOrder;
        this.IsEnabled = entity.IsEnabled;
    }
    #endregion

    #region Methods
    /// <summary>
    /// Explicit conversion to entity.
    /// </summary>
    /// <param name="model"></param>
    public static explicit operator Entities.Tag(TagModel model)
    {
        var entity = new Entities.Tag(model.Id, model.Name, model.Code)
        {
            Id = model.Id,
            Name = model.Name,
            Code = model.Code,
            Description = model.Description,
            IsEnabled = model.IsEnabled,
            SortOrder = model.SortOrder,
            Version = model.Version ?? 0,
        };
        return entity;
    }
    #endregion
}
